WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift

```diff
@@ -26,10 +26,12 @@ final class OrdersTests: XCTestCase {
     }
 
     func test_create_new_order() throws {
+        let products = try GetMocks.readProductsData()
+
         try TabNavComponent().goToOrdersScreen()
             .startOrderCreation()
-
-        try NewOrderFlow.editOrderStatus()
+            .editOrderStatus()
+            .addProduct(byName: products[0].name)
             .createOrder()
```

Comment:
I know that this is also not introduced in this PR, but noticed that the test don't have a verification step (e.g. [verifyOrdersScreenLoaded()](https://github.com/woocommerce/woocommerce-ios/blob/16dc8f30c85084bfe0c209fd124d137639e50577/WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift#L25) on the `testOrdersScreenLoads()` test). 

We should aim for at least one verification for the test, for this we could add one at the end of the test to ensure that the test lands on the order screen with the expected values (in this case, we could use the validate the product name and order status).

Reply:
Thanks for bringing this up! Verifying that the screen loads is definitely an important part of the test.
On iOS, the way our screen objects are written includes this verification. 
At each step of the test (whenever we navigate), the screen in question is verified in the `super.init` statement, which relies on the `init()` in ScreenObject.

This `init()` statement checks that the screen "self" is loaded. It calls `waitForScreen()` which checks that `isEnabled == true` for each expected element getter.

> we could add one at the end of the test to ensure that the test lands on the order screen with the expected values

Our goal with this test is to create a minimal functional test, without checking data (https://github.com/woocommerce/woocommerce-ios/issues/6524). That's partly for ease of implementation, but also because this data is checked thoroughly at the Unit Test level.

We're trying this approach in order to allow developers to involved with the UI testing easily, but we can always extend the test if desired!

Reply:
Thanks for linking https://github.com/woocommerce/woocommerce-ios/issues/6524 to this! That makes it much clearer what the team strategy is for UI tests.

Though my recommendation would still be to update it later so that the assertion is clear from the test itself, like how unit tests are using `Given, When, Then`s, and other UI tests are using `verify...` in this test, it looks like there's no `Then`/assertion until we look into the return value of `createOrder()`